### ISSUE TITLE
Feature: Detect Markdown table syntax in AsciiDoc files

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#38]
## Problem

AsciiDoc files may accidentally contain Markdown table syntax instead of proper AsciiDoc tables. This commonly happens when documentation is AI-generated or when contributors are more familiar with Markdown.

**Example of incorrect Markdown syntax in AsciiDoc:**

```asciidoc
| Dimension | Score | Level | Evidence |
|-----------|-------|-------|----------|
| Code Type | 2 | Business Logic | Click commands... |
| Language | 2 | Dynamically typed | Python 3.12+ |
```

**Correct AsciiDoc syntax:**

```asciidoc
[cols="2,1,2,4"]
|===
| Dimension | Score | Level | Evidence

| Code Type | 2 | Business Logic | Click commands...
| Language | 2 | Dynamically typed | Python 3.12+
|===
```

## Current Behavior

The linter does NOT detect Markdown table syntax (pipes with dashes `|---|---|`).

Testing with `asciidoc-linter file.adoc` on a file containing the Markdown table above produces no warnings about the table.

## Expected Behavior

The linter should detect and report Markdown table syntax as a warning or error.

## Proposed Solution

Add a new rule (e.g., **FMT005: Markdown table syntax**) that:

1. Detects lines matching the Markdown table separator pattern: `|---` or `| --- |` etc.
2. Detects Markdown table header/data rows (lines starting and ending with `|`)
3. Reports these as violations with a message like:
   - "Markdown table syntax detected. Use AsciiDoc table syntax instead."
   - Suggest the correct AsciiDoc format in the error message

## Implementation Notes

Detection regex patterns:
- Table separator: `^\s*\|[-:\s|]+\|?\s*$` (matches `|---|---|` or `| --- | --- |`)
- Table row: `^\s*\|.*\|\s*$` (lines starting/ending with `|`)
- Combined: If separator row is found, flag all adjacent rows matching table row pattern

Rule severity: **error** (this breaks AsciiDoc rendering in most processors)

## Use Case

We discovered this issue when AI-generated Architecture Decision Records (ADRs) contained Markdown tables. The linter did not catch them, and they were merged. We want to prevent this in the future by adding the linter as a pre-commit hook.

## Related

- This would complement the existing FMT004 rule for Markdown syntax detection
- Could be part of a broader Markdown syntax detection suite

## Priority

High - This is a common error that breaks documentation rendering.
